src/hooks/useAddLiquidityHandler.tsx

```diff
@@ -158,72 +169,85 @@ export const useAddLiquidityHandler = () => {
-            let res = await tx.wait()
-            const { args } = res.events[0]
-            let message = 'Add Liquidity Success: ' + args.pool + args.reservesLen.toNumber() + args.shares.toNumber()
-            notifyDismiss(loading)
-            notifySuccess(message)
-          } else {
-            notifyDismiss('Waiting for transaction to complete')
+    if (!posManager || !accountInfo) {
+      notifyError('Please connect wallet.')
+      return
+    }
+
+    if (!ethers.utils.isAddress(cfmmPoolAddr)) {
+      notifyError('Selected pair is not a valid cfmm pool pair.')
+      return
+    }
+
+    if (accountInfo.address && provider) {
+      let tokenAContract = new ethers.Contract(
+        tokenASelected.address,
+        IERC20.abi,
+        accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+      )
+
+      let approvalA = await validateAllowance(
+        accountInfo.address,
+        tokenAContract,
+        Number(tokenAInputVal),
+        POSITION_MANAGER_ADDRESS || ''
+      )
+
+      let tokenBContract = new ethers.Contract(
+        tokenBSelected.address,
+        IERC20.abi,
+        accountInfo && accountInfo?.address ? provider.getSigner(accountInfo?.address) : provider
+      )
+
+      let approvalB = await validateAllowance(
+        accountInfo.address,
+        tokenBContract,
+        Number(tokenBInputVal),
+        POSITION_MANAGER_ADDRESS || ''
+      )
+
+      let loading = notifyLoading('Waiting for transaction to complete')
+      try {
+        if (approvalA && approvalB) {
+          const DepositReservesParams = {
+            cfmm: cfmmPoolAddr,
+            amountsDesired: [BigNumber.from(parseFloat(tokenAInputVal)), BigNumber.from(parseFloat(tokenBInputVal))],
+            amountsMin: [0, 0],
+            to: accountInfo.address,
+            protocol: 1,
+            deadline: ethers.constants.MaxUint256,
           }
-        } catch (e: any) {
+          let tx = await posManager.depositReserves(DepositReservesParams, {
+            gasLimit: process.env.NEXT_PUBLIC_GAS_LIMIT,
+          })
+          let res = await tx.wait()
+          const { args } = res.events[0]
+          let message = 'Add Liquidity Success: ' + args.pool + args.reservesLen.toNumber() + args.shares.toNumber()
           notifyDismiss(loading)
-          if (e?.code === 'ACTION_REJECTED') {
-            notifyError('User rejected the transaction')
-          } else {
-            notifyError('An error occurred while adding LP Token Liquidity. Please try again')
-          }
+          notifySuccess(message)
+        } else {
+          notifyDismiss('Waiting for transaction to complete')
+        }
+      } catch (e: any) {
+        notifyDismiss(loading)
+        if (e?.code === 'ACTION_REJECTED') {
+          notifyError('User rejected the transaction')
+        } else {
+          notifyError('An error occurred while adding LP Token Liquidity. Please try again')
         }
```

Comment:
isn't this error message supposed to be for the `addLpLiquidity` function?

Reply:
Changed to be reserves specific message.